imagebuildah/stage_executor.go

```diff
@@ -421,13 +421,20 @@ func (s *StageExecutor) performCopy(excludes []string, copies ...imagebuilder.Co
 				copiesExtend = append(copiesExtend, copyEntry)
 			}
 			copySources := []string{}
+			// Create seperate subdir for processing all heredoc instructions
+			// in this stage.
```

Comment:
The syntax doesn't inherently prevent the basename from being used again with a different destination or command within the same stage, we might as well use a separate directory for each file before someone tries it.